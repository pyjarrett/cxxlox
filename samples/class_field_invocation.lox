class FieldInvocation {
    init() {
        fun defaultCallback() {
            print "Default callback";
        }
        this.callback = defaultCallback;
    }

    method() {
        print "A method call";
    }
}

fun anotherCallback() {
    print "Another callback";
}

var fi = FieldInvocation();
fi.callback();
fi.callback = anotherCallback;
fi.callback();
fi.method();