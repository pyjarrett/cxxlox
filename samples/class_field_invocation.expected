Default callback
Another callback
A method call
